python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -133,18 +133,53 @@ def __new__(cls, asset_key: AssetKey, asset_backfill_status: Optional[AssetBackf
 
 
 @whitelist_for_serdes(field_serializers={"backfill_start_time": DatetimeFieldSerializer})
-class AssetBackfillData(NamedTuple):
-    """Has custom serialization instead of standard Dagster NamedTuple serialization because the
-    asset graph is required to build the AssetGraphSubset objects.
-    """
-
-    target_subset: AssetGraphSubset
-    requested_runs_for_target_roots: bool
-    latest_storage_id: Optional[int]
-    materialized_subset: AssetGraphSubset
-    requested_subset: AssetGraphSubset
-    failed_and_downstream_subset: AssetGraphSubset
-    backfill_start_time: datetime
+class AssetBackfillData(
+    NamedTuple(
+        "_AssetBackfillData",
+        [
+            ("target_subset", AssetGraphSubset),
+            ("requested_runs_for_target_roots", bool),
+            ("latest_storage_id", Optional[int]),
+            ("materialized_subset", AssetGraphSubset),
+            ("requested_subset", AssetGraphSubset),
+            ("failed_and_downstream_subset", AssetGraphSubset),
+            ("backfill_start_time", datetime),
+            ("partitions_def_ids_by_asset_key", Optional[Mapping[AssetKey, str]]),
```

Comment:
Why exactly do we need this? Because we want to fail the backfill if the partitions def changes during its execution?

Is this definitely still necessary now?

Reply:
> Because we want to fail the backfill if the partitions def changes during its execution?

Yes

The alternative would be checking if each targeted partition is still existent in the partitions def